pywbem/_subscription_manager.py

```diff
@@ -343,30 +343,38 @@ def add_server(self, server):
         self._owned_subscriptions[server_id] = []
         self._owned_filters[server_id] = []
         self._owned_destinations[server_id] = []
+        # self._systemnames[server_id] = None
+
+        # TODO expand this to account for WBEM Server profile.
 
         # Recover any owned destination, filter, and subscription instances
         # that exist on this server
-
-        this_host = getfqdn()
-
         dest_name_pattern = re.compile(
-            _format(r'^pywbemdestination:owned:{0}:{1}:[^:]*$',
-                    this_host, self._subscription_manager_id))
+            _format(r'^pywbemdestination:owned:{0}:[^:]*$',
+                    self._subscription_manager_id))
         dest_insts = server.conn.EnumerateInstances(
             DESTINATION_CLASSNAME, namespace=server.interop_ns)
-        for inst in dest_insts:
-            if re.match(dest_name_pattern, inst.path.keybindings['Name']) \
-                    and inst.path.keybindings['SystemName'] == this_host:
-                self._owned_destinations[server_id].append(inst)
+        if dest_insts:
+            self._systemnames[server_id] = dest_insts[0]['SystemName']
+            svr_name = self._systemnames[server_id]
+            for inst in dest_insts:
+                if re.match(dest_name_pattern, inst.path.keybindings['Name']) \
+                        and inst.path.keybindings['SystemName'] == svr_name:
+                    self._owned_destinations[server_id].append(inst)
 
         filter_name_pattern = re.compile(
-            _format(r'^pywbemfilter:owned:{0}:{1}:[^:]*:[^:]*$',
-                    this_host, self._subscription_manager_id))
+            _format(r'^pywbemfilter:owned:{0}:[^:]*:[^:]*$',
+                    self._subscription_manager_id))
```

Comment:
As discussed in the call, the subscription mgr ID is unique on the client host, so the client host should be put back in, and also be compared against.

Reply:
DONE